FirebaseAdmin/FirebaseAdmin/Messaging/Util/AnalyticsLabelChecker.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace FirebaseAdmin.Messaging.Util
+{
+    /// <summary>
+    /// Checker for analytics label.
+    /// </summary>
+    public static class AnalyticsLabelChecker
+    {
+        private static ImmutableHashSet<char> alowedChars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-~%".ToCharArray().ToImmutableHashSet();
+
+        /// <summary>
+        /// Checks anytytics labels an throw if not valid.
+        /// </summary>
+        /// <exception cref="ArgumentException">If analytics label does not match pattern.</exception>
+        /// <param name="analyticsLabel">Analytics label.</param>
+        public static void CheckAnalyticsLabelOrThrow(string analyticsLabel)
```

Comment:
Nit: Perhaps just call it `ValidateAnalyticsLabel()`. I don't feel too strongly about it, so your call.